### DISCUSSION TITLE
如何传递function类型的prop到子组件？

### DISCUSSION THREAD

**Discussion by maopixin** [NervJS/taro#11412] [Q & A]
```code
<van-datetime-picker
  type="date"
  :value="startTime"
  :min-date="minDate"
  title="选择开始时间"
  confirm-button-text="确定"
  toolbar-class="timepickerToolBar"
  @confirm="onStartTimePick"
/>
```
在使用vant weapp时传递
formatter | 选项格式化函数 | (type, value) => value 
时，在子组件中取到formatter是null，但是传递布尔值之类的是可以正常传递的。
想知道应该怎么做，才能传递函数类型的prop

**Comment by smoothdvd** [comment]
同问，在用react编写小程序时（使用WeUI扩展的组件）也有这个问题。

**Comment by TheKonka** [comment]
```js
formatter = {{  fn:(type, value) => value} }
```

我之前 react 里面引用支付宝原生组件是这样写的， 不知道 可不可以

**Reply by smoothdvd** [reply]
@konka-king 试了一下，不行。这样写prop的类型就变成object而不是function了。

**Comment by Chen-jj** [comment]
需要 hack 以绕过，详细写法和讨论请看下方 issues，已补充进文档

[React 写法](https://github.com/NervJS/taro/issues/8495#issuecomment-1179526364)

[Vue3 写法](https://github.com/NervJS/taro/issues/10337#issuecomment-989834523)
